### ISSUE TITLE
Accessibility-> Search page doesn't retain the search turn when returning back to it from a site page

### ISSUE THREAD

**Issue by manuji** [bcgov/nr-rec-resources#661]
**Describe the bug**
On Find a site or trail page, after a successful search if user clicks on a site page to view the site page and navigate back to the Find a site or trail page, the previous search has been reset. This issue fails the [3.3.7 Redundant Entry (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry) criterion. 

**To Reproduce**
1. Go to 'https://d1vrz2gt4nmfe5.cloudfront.net/search'
2. Search for a keyword (eg: Tree)
3. Click on a site from the search results
4. From the Site page try navigating back to the Find a site or trail page either by breadcrumbs or browser back button
5. Check if the search term and the search results retain on the page. 

**Expected behavior**
When navigating back to the Find a site or trail page, the search term and the results should retail. 

**Screenshots**

https://github.com/user-attachments/assets/024f0139-ff81-4d19-adda-01437bab7bac

**Additional context**
Tested on MacOS/ Chrome.
